libbeat/outputs/logstash/config.go

```diff
@@ -11,6 +11,16 @@ type logstashConfig struct {
 	CompressionLevel int                `config:"compression_level"`
 	MaxRetries       int                `config:"max_retries"`
 	TLS              *outputs.TLSConfig `config:"tls"`
+	SOCKSProxy       *socksProxyConfig  `config:"socks_proxy"`
+}
+
+// socksProxyConfig holds the configuration information required to proxy
+// Logstash connections through a SOCKS5 server.
+type socksProxyConfig struct {
+	HostPort  string `config:"hostport"`   // host:port of the SOCKS5 server.
```

Comment:
maybe change config to 'host' + add ability to set username/password right in address (like we do support for other  outputs)